### ISSUE TITLE
[native-image] Failing with Type not found during analysis: HotSpotType<Lcom/sun/proxy/jdk/proxy3/$Proxy324;, resolved>

### ISSUE THREAD

**Issue by aclement** [oracle/graal#1921]
Upgrading a Spring MVC sample app (with tomcat) to graal 19.3 from 19.2. Have this error not sure what to do (on mac mojave 10.14.5):
```
[tc:26321]      compile:  67,932.30 ms
[tc:26321]        image:     916.14 ms
Fatal error: com.oracle.svm.core.util.VMError$HostedError: com.oracle.graal.pointsto.util.AnalysisError$TypeNotFoundError: Type not found during analysis: HotSpotType<Lcom/sun/proxy/jdk/proxy3/$Proxy324;, resolved>
	at com.oracle.svm.core.util.VMError.shouldNotReachHere(VMError.java:70)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:636)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:445)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: com.oracle.graal.pointsto.util.AnalysisError$TypeNotFoundError: Type not found during analysis: HotSpotType<Lcom/sun/proxy/jdk/proxy3/$Proxy324;, resolved>
	at com.oracle.graal.pointsto.util.AnalysisError.typeNotFound(AnalysisError.java:134)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.createType(AnalysisUniverse.java:221)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.lookupAllowUnresolved(AnalysisUniverse.java:205)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.lookup(AnalysisUniverse.java:182)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.lookup(AnalysisUniverse.java:75)
	at com.oracle.graal.pointsto.infrastructure.UniverseMetaAccess$1.apply(UniverseMetaAccess.java:52)
	at com.oracle.graal.pointsto.infrastructure.UniverseMetaAccess$1.apply(UniverseMetaAccess.java:49)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at com.oracle.graal.pointsto.infrastructure.UniverseMetaAccess.lookupJavaType(UniverseMetaAccess.java:84)
	at com.oracle.graal.pointsto.meta.AnalysisMetaAccess.lookupJavaType(AnalysisMetaAccess.java:47)
	at com.oracle.svm.hosted.ameta.HostedDynamicHubFeature.replace(HostedDynamicHubFeature.java:52)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.replaceObject(AnalysisUniverse.java:534)
	at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.replaceObject(AnalysisConstantReflectionProvider.java:201)
	at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.interceptValue(AnalysisConstantReflectionProvider.java:172)
	at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readValue(AnalysisConstantReflectionProvider.java:101)
	at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readFieldValue(AnalysisConstantReflectionProvider.java:77)
	at com.oracle.svm.hosted.meta.HostedField.readValue(HostedField.java:156)
	at com.oracle.svm.hosted.image.NativeImageHeap.addObjectToBootImageHeap(NativeImageHeap.java:420)
	at com.oracle.svm.hosted.image.NativeImageHeap.addObject(NativeImageHeap.java:278)
	at com.oracle.svm.hosted.image.NativeImageHeap.processAddObjectWorklist(NativeImageHeap.java:527)
	at com.oracle.svm.hosted.image.NativeImageHeap.addTrailingObjects(NativeImageHeap.java:112)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:619)
	... 7 more
Error: Image build request failed with exit status 1
```

**Comment by aclement** [comment]
This was graal 19.3 (Java11)

**Comment by aclement** [comment]
Some repro steps:
```
git clone git@github.com:spring-projects-experimental/spring-graal-native.git
git checkout upgrade_19_3
cd spring-graal-native/spring-graal-native-feature
mvn clean package
cd ../spring-graal-native-samples/springmvc-tomcat
./compile.sh
```

**Comment by dougxc** [comment]
May be related to #1921.

**Comment by dougxc** [comment]
May be related to #1927 .

**Comment by eginez** [comment]
@aclement Is this happening with java 8 or java 11?

**Comment by eginez** [comment]
This seems to have been fixed a5aa34b as well. I can no longer replicate this issue building from master
